cas/service.c

```diff
@@ -1353,16 +1405,28 @@ static int cas_fom_tick(struct m0_fom *fom0)
 		 *
 		 */
 		if (!m0_dtm0_tx_desc_is_none(&cas_op(fom0)->cg_txd)) {
-			/*
 			struct m0_be_tx_credit dtm0logrec_cred;
-
-			m0_be_dtm0_log_credit(M0_DTML_EXECUTED,
-					      tx,
-					      cas_op(fom0)->cg_txd,
+			void       *buf = NULL;
+			m0_bcount_t len = 0;
+			int	    rc;
+
+			/* TDB: while calculating the credits we need only the
+			   size of the payload. Check if m0_be_dtm0_log_credit
+			   needs to be updated to accept a size instead of the
+			   actual payload. */
+
+			rc = m0_xcode_obj_enc_to_buf(&M0_XCODE_OBJ(m0_cas_op_xc,
+						      cas_op(fom0)), &buf, &len);
+			M0_ASSERT(rc == 0);
+
+			m0_be_dtm0_log_credit(M0_DTML_PERSISTENT,
+					      &cas_op(fom0)->cg_txd,
+					      buf,
```

Comment:
The `void *` vs. `m0_buf *` problem here. Credit calculation may not work as we expect because the nob field is not filled correctly. Let's define `buf` as `m0_buf`.

Reply:
ok.